### PR TITLE
[FIX] Allow tokenizer to parse numbers greater than INT_MAX.

### PR DESCRIPTION
**Pull request by tkonolige**
I've also improved the error handling.

@jroesch @masahi @tqchen

### CODE REVIEW

**Comment by tqchen** [comment]
thanks @tkonolige perhaps we can introduce a regression testcase(e.g. parse a long int) ?

File: src/parser/parser.cc [hunk 1]
```diff
@@ -524,12 +524,12 @@ class Parser {
...
       DLDevice dev = {DLDeviceType::kDLCPU, 0};
-      auto dtype = String2DLDataType("int32");
+      auto dtype = String2DLDataType("int64");
       auto data = NDArray::Empty({}, dtype, dev);
-      auto array = reinterpret_cast<int32_t*>(data->data);
+      auto array = reinterpret_cast<int64_t*>(data->data);
       // revisit this, literal node issue.
       int64_t value = Downcast<tvm::Integer>(token->data);
-      array[0] = (int32_t)value;
+      array[0] = (int64_t)value;
```
Comment (tqchen) [inline src/parser/parser.cc:532]: `static_cast<int64_t>`(always prefer c++ style cast)

Comment (altanh) [reply src/parser/parser.cc:532]: it looks like we don't even need this particular cast now

**Comment by tkonolige** [comment]
@tqchen @altanh Can we merge?

**Review by tqchen — APPROVED** [review]

**Pull request merged**